sdk/identity/Azure.Identity/src/DefaultAzureCredential.cs

```diff
@@ -28,8 +28,8 @@ namespace Azure.Identity
     /// </remarks>
     public class DefaultAzureCredential : TokenCredential
     {
-        private const string DefaultExceptionMessage = "The DefaultAzureCredential failed to retrieve a token from the included credentials.";
-        private const string UnhandledExceptionMessage = "The DefaultAzureCredential failed due to an unhandled exception: ";
+        private const string DefaultExceptionMessage = "DefaultAzureCredential failed to retrieve a token from the included credentials.";
```

Comment:
What do you mean by "included credentials"?

Reply:
So this message get's augmented with unavailable exception messages from the credentials which were included, so the final message is something like this:

>> DefaultAzureCredential failed to retrieve a token from the included credentials. ( EnvironmentCredential authentication unavailable, environment variables are not fully configured. ManagedIdentityCredential not available, no managed identity endpoint found. SharedTokenCacheCredential authentication unavailable, no accounts were found in the cache. )

Given that do you still feel that this wording is confusing?

Reply:
How about this?

Before:
```
( EnvironmentCredential authentication unavailable, environment variables are not fully configured. ManagedIdentityCredential not available, no managed identity endpoint found. SharedTokenCacheCredential authentication unavailable, no accounts were found in the cache. )
```

After:
```
DefaultAzureCredential was unable to retrieve a token.
EnvironmentCredential: Environment variables not fully configured. 
ManagedIdentityCredential: No Managed Identity endpoint found.
SharedTokenCacheCredential: No accounts found in cache.
```

Reply:
Something to consider in this is that the error message would need to make sense on it's own as well.

For example, if I do `var cred = new EnvironmentCredential()` and it's not configured what will the message be?

Reply:
So the fact that the error messages need to stand on their own is the reason they are worded the way they are. But I can format them with newlines as you suggest.
 
~~~
 DefaultAzureCredential failed to retrieve a token from the included credentials.
 EnvironmentCredential authentication unavailable, environment variables are not fully configured. 
 ManagedIdentityCredential not available, no managed identity endpoint found. 
 SharedTokenCacheCredential authentication unavailable, no accounts were found in the cache.
~~~

Does this work for you?

Reply:
I personally think they stand fine on their own as I had them above.  

1. It's an exception message, so they already know it didn't work.
2. Less text to read


I don't see "authentication unavailable" adding more to the reason that comes right after it.  If you do decide to include it, then please have them be consistent...Managed has not available, the others have unavailable.